src/uri.cr

```diff
@@ -265,8 +375,15 @@ class URI
         return i
       end
 
-      io.write_byte (first_num * 16 + second_num).to_u8
+      encoded = (first_num * 16 + second_num).to_u8
       i += 1
+      if yield encoded
```

Comment:
Why only here? Why yield the 8-bit truncated codepoint (if I follow correctly) and refer to it as byte? Why not let the caller decide which bytes to write if we already do this dance?